### PR TITLE
Fcrepo 2105 final

### PR DESCRIPTION
**Pull request by dbernstein**
https://jira.duraspace.org/browse/FCREPO-2105

### CODE REVIEW

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/StreamUtils.java [hunk 1]
```diff
@@ -20,16 +20,40 @@
...
  * @author acoburn
  * @since February 10, 2016
  */
 public class StreamUtils {
 
+    private static final Logger LOGGER = getLogger(StreamUtils.class);
+
+    private static Boolean enableParallel = false;
+
+    private static final String FCREPO_ENABLE_PARALLEL_STREAMING_BY_DEFAULT_KEY =
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/StreamUtils.java]: Let's simplify this to `fcrepo.streaming.parallel`

Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/StreamUtils.java]: Once the PR is in the codebase, we should also update the documentation with this configuration element.
https://wiki.duraspace.org/display/FEDORA4x/Configuration+Options+Inventory

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java [hunk 2]
```diff
@@ -128,7 +130,21 @@ private static void write(final RdfStream rdfStream,
...
             stream.start();
             nsPrefixes.forEach(stream::prefix);
-            rdfStream.forEach(stream::triple);
+
+            // ensure that writes to the stream object are serialized
+            // in case the rdfStream object is running in parallel mode.
+            final ReentrantLock lock = new ReentrantLock();
+            rdfStream.forEach(x -> {
+                try {
+                    lock.tryLock(30, TimeUnit.SECONDS);
```
Comment (ajs6f) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: If you're just trying to keep a single thread at a time running through the `StreamRDF`, why not just subtype it with `synchronize void triple(Triple t)`? No need to explicitly manage a lock.

Comment (dbernstein) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: @ajs6f :  This seems like a good idea to me.  @awoods:  any reason not to go the subclassing route?

Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: @dbernstein : agreed. Go for it.

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/StreamUtils.java (hunk 1 above)
Comment (dbernstein) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/StreamUtils.java]: okay - yes re simplification. And yes I will update the documentation.

**Comment by awoods** [comment]
@dbernstein : as a process comment, in order to facilitate transparency in what a pull-request is designed to address, the subject text of the pull-request should provide a brief, but meaningful description:
https://wiki.duraspace.org/display/FF/Git+Workflow#GitWorkflow-Committing

`Fcrepo 2105 final` requires an understanding of what that JIRA ticket is about.

That said, you rock :metal: ! Keep the great work flowing!

**Comment by awoods** [comment]
hmm... I am still getting the same test failures:
```
Failed tests: 
  ReferencesRdfContextTest.testStrongReferences:123 null
  ReferencesRdfContextTest.testWeakReferences:135 null
```

..and adding a assertion error messages shows that the model is empty:
```
assertTrue(model.contains(createResource("info:fedora/b"),
                createProperty("info:strong"),
                createResource("info:fedora/a")));
```
results in:
```
Failed tests: 
  ReferencesRdfContextTest.testStrongReferences:123 model: <ModelCom   {} | >
  ReferencesRdfContextTest.testWeakReferences:135 model: <ModelCom   {} | >
```

**Comment by dbernstein** [comment]
I'll take a look.

**Comment by dbernstein** [comment]
@awoods : So confirmed,  I was mistaken: the performance improvements were lost with the latest commit.   I've reverted locally and am working on another fix to the broken unit tests.

**Comment by awoods** [comment]
This looks good to me. 
@escowles ? @ruebot ? would you mind giving this a look and squash/merging if you are happy with the PR. I would do it myself if some of my commits were not intermingled.

**Pull request merged**